src/olympia/versions/models.py

```diff
@@ -232,11 +232,80 @@ def should_have_due_date(self, negate=False):
             needshumanreview__is_active=True,
             needshumanreview__reason=NeedsHumanReview.REASONS.DEVELOPER_REPLY,
         )
-        return (
+        qs = (
             method(is_needs_human_review | is_pre_review_version | has_developer_reply)
             .using('default')
             .distinct()
         )
+        if annotate_reasons:
+            nhr_q = self.filter(is_needs_human_review, id=OuterRef('pk'))
+            qs = qs.annotate(
+                _due_date_reasons=Case(
+                    When(
+                        Exists(
+                            nhr_q.filter(
+                                needshumanreview__reason=NeedsHumanReview.REASONS.CINDER_ESCALATION
+                            )
+                        ),
+                        then=Version.DUE_DATE_REASON_NHR_CINDER_ESCALATION,
+                    ),
+                    default=0,
+                )
+                + Case(
+                    When(
+                        Exists(
+                            nhr_q.filter(
+                                needshumanreview__reason=NeedsHumanReview.REASONS.ABUSE_ADDON_VIOLATION
+                            )
+                        ),
+                        then=Version.DUE_DATE_REASON_NHR_ABUSE,
+                    ),
+                    default=0,
+                )
```

Comment:
fwiw this can be rewritten as 
```suggestion
                + (
                    Exists(
                        nhr_q.filter(
                            needshumanreview__reason=NeedsHumanReview.REASONS.ABUSE_ADDON_VIOLATION
                        ),
                        output_field=models.IntegerField()
                    ) * Version.DUE_DATE_REASON_NHR_ABUSE
                )
```
but I'm not convinced it's more readable than "Case When Exists q then v default 0"